### PR TITLE
Style/RescueModifier doesn't detect offence in single-line method

### PR DESCRIPTION
**Pull request by urbanautomaton**
Currently the cop Style/RescueModifier does not detect an offence for the following, which I would expect to fail:

```ruby
def a_method
  test rescue nil
end
```

Here is [output from an example file](https://gist.github.com/urbanautomaton/4ef5aaf6e4ea9638f09b).

The issue seems to be that the method body is detected as a node of type `:rescue`, causing it to be [ignored by this line](https://github.com/urbanautomaton/rubocop/blob/ea1feb6351014e21efc2d6b4d450b5208566ffd4/lib/rubocop/cop/style/rescue_modifier.rb#L32). It's not clear to me how to fix this without breaking other test cases, though, as `parser` returns an identical AST for the following code, which should not register an offence:

```ruby
def another_method
  test
rescue
  nil
end
```

In the absence of any better ideas, I've added a failing test case. Any suggestions on how to make it pass would be gratefully received. :smile:

### CODE REVIEW

**Comment by bbatsov** [comment]
If this AST is the same, this is like a `parser` bug. //cc @whitequark

**Comment by whitequark** [comment]
Nope, not a bug. Parser ASTs are semantic, and these constructs are semantically equivalent.

**Comment by whitequark** [comment]
One thing you can do is look at the token stream, where block rescue would be `:kRESCUE`, but modifier rescue would be `:kRESCUE_MOD`.

**Comment by bbatsov** [comment]
Yep, perhaps this would be best.

**Comment by urbanautomaton** [comment]
@whitequark: Thanks, that did the trick nicely.

@bbatsov: I've implemented the fix and all the tests now pass. I had to update the HTML formatter output fixture because of a minor order change, I presume because the offence is now reported on the `rescue` token rather than the whole expression. If there's a way to preserve the existing order, let me know and I'll update as necessary.

**Comment by bbatsov** [comment]
@urbanautomaton Looks good. Just add some test about the position highlighting and squash the commits together.

**Comment by bbatsov** [comment]
Add a changelog entry as well.

**Comment by urbanautomaton** [comment]
Done, done and done.

I assume since I've not introduced any new uncovered lines of code that the coverage decrease is either a) noise or b) because I've slightly reduced the number of covered lines.

File: spec/rubocop/cop/style/rescue_modifier_spec.rb [hunk 1]
```diff
@@ -13,6 +13,13 @@
       .to eq(['Avoid using `rescue` in its modifier form.'])
   end
 
+  it 'locates the rescue within the line' do
+    inspect_source(cop,
+                   'method rescue handle')
+    expect(cop.offenses.size).to eq(1)
+    expect(cop.offenses.first.location.source).to eq('rescue')
```
Comment (bbatsov) [inline spec/rubocop/cop/style/rescue_modifier_spec.rb]: Checkout `cop.highlights`. It's the preferred way to check for offence highlights.

Comment (urbanautomaton) [inline spec/rubocop/cop/style/rescue_modifier_spec.rb]: Cool, changed.

**Comment by bbatsov** [comment]
:+1: Thanks!

**Pull request merged**